eventsourcing/domain/services/eventstore.py

```diff
@@ -106,7 +106,7 @@ def append(self, new_stored_event, new_version_number=None, max_retries=3, artif
         Saves given stored event in this repository.
         """
         # Check the new event is a stored event instance.
-        assert isinstance(new_stored_event, StoredEvent)
```

Comment:
This was the pain point because the "Abstract" class has default implementation coded in. To use a different StoredEvent, I had to copy&paste code to my sub-class and override the whole function.